worlds/tracker/docs/apworld-integration.md

```diff
@@ -0,0 +1,98 @@
+        if "starting_location" in slot_data:
+            self.origin_region_name = slot_data["starting_location"]
+
+        if "entrances" in slot_data:
+            # Update entrance connections for ER
+            entrances = {
+                entrance.name: entrance
+                for region in self.get_regions()
+                for entrance in region.entrances
+            }
+            for source_exit, target_entrance in slot_data["entrances"]:
+                entrances[source_exit].connected_region = entrances[target_entrance].parent_region
+```
+
+You may also want to adjust your world's behaviour in other functions to reduce the amount of work you need to do in `interpret_slot_data`. To do so, you can check the `generation_is_fake` attribute on the `multiworld` object, which will be `True` if your world is generating inside UT. You can use this to skip doing things that will be handled by `interpret_slot_data` later, or to run code that would normally be skipped based on options. For example, you can create every possible location and let UT filter them out to just what exists from the server so you don't need to worry about creating them later.
+
+```python
+    def create_regions(self) -> None:
+        is_ut = getattr(self.multiworld, "generation_is_fake", False)
+        if self.randomly_rolled_property == 1 or is_ut:
+            # This location will always be created by UT regardless of what UT happens to roll
+            self.create_location(...)
+```
+
+## Generating without a YAML
+
+You can also make it so a YAML is not required to generate for your world. To do so, you must first store all options that affect generation in your slot data.
+
+```python
+    def fill_slot_data(self) -> dict[str, Any]:
+        # Store all defined options
+        option_fields = [field.name for field in dataclasses.fields(self.options)]
+        return {
+            # Keep other randomized results as above
+            "options": self.options.as_dict(*option_fields),
+        }
+```
+
+To make implementation simpler, instead of the large amount of adjustments you'd likely have to make in `interpret_slot_data`, you can instead turn it into a static method that simply returns the slot data. This will inform UT that instead of doing an initial generation when it opens, it will do a regeneration once connected to the multiworld.
+
+```python
+    @staticmethod
+    def interpret_slot_data(slot_data: dict[str, Any]) -> dict[str, Any]:
+        # Trigger a regen in UT
+        return slot_data
+```
+
+During the regeneration, you can access the slot data via `re_gen_passthrough` on the multiworld object. Early in the generation you can inspect this object to set any instance attributes or options needed from the slot data so your later functions will behave as expected.
+
+```python
+    def generate_early(self) -> None:
+        re_gen_passthrough = getattr(self.multiworld, "re_gen_passthrough", {})
```

Comment:
```suggestion
        re_gen_passthrough = getattr(self.multiworld, "re_gen_passthrough", False)
```
Earlier you use False as the default, so it should stay consistent.

Reply:
That was `generation_is_fake` that defaulted to False, this is `re_gen_passthrough`, which is a dict.